packages/cli/src/cmds/beacon/handler.ts

```diff
@@ -33,6 +34,18 @@ const HOURS_TO_MS = 3600 * 1000;
  * Runs a beacon node.
  */
 export async function beaconHandler(args: BeaconArgs & GlobalArgs): Promise<void> {
+  // must be EARLY in startup because may change UV_THREADPOOL_SIZE
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let invalidPoolSizeValue: any;
+  let poolSizeIsNanWarn = false;
+  const defaultThreadpoolSize = os.availableParallelism();
```

Comment:
why do we need to set this in multiple places?

Reply:
There is no way to pass the warning through from preInitialization to the handler where the logger is available. Didn't want to use console. Log.

Reply:
There is no way to pass the warning through from preInitialization to the handler where the logger is available. Didn't want to use console. Log.

Reply:
But why is the env set here again? As I understand is should be set in preInitialization, and here we just want to log a warnings?

Reply:
There is no way to pass the fact that it was reset (and needing a warning) from the initialization into the handler where the logger is.  I found that yargs has a `context` object but there wasnt clear directions on how that gets passed into the handler.  Looking a bit deeper now